src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -641,6 +641,7 @@ def execute(self, args):
 
             if hasattr(expanded_arg, '_subscription'):
                 cmd_copy.cli_ctx.data['subscription_id'] = expanded_arg._subscription  # pylint: disable=protected-access
+                self.cli_ctx.data['subscription_id'] = expanded_arg._subscription  # pylint: disable=protected-access
```

Comment:
This line is added because:

L643 will only add `subscription_id` info to `AzCliCommand.cli_ctx`;
`AzCommandGroup.cli_ctx` and `AzCommandsLoader.cli_ctx` was not updated which is needed in creating client factory

Reply:
We shouldn't change this line as it works fine for other command modules. 

Instead, change 

https://github.com/Azure/azure-cli/blob/20c5d089010ebbcdf1c7c09409b3a51cb7494f64/src/azure-cli/azure/cli/command_modules/keyvault/_command_type.py#L100

to 

```py
client = client_factory(command_args['cmd'].cli_ctx, command_args)
```

so that `AzCliCommand.cli_ctx` is used.

Reply:
Good point! Modified~